### PR TITLE
Mes 3020 search tests to rekey

### PR DESCRIPTION
**Pull request by szabikr**
Creating RekeySearchEffect which will deal with the RekeySearchProvider to get the booked test slot from the lambda. Eventually store the testSlot data to the store and reselect to the RekeySearchPage.

### CODE REVIEW

File: src/app/app.module.ts [hunk 1]
```diff
@@ -103,6 +104,7 @@ const enableDevTools = environment && environment.enableDevTools;
     { provide: HAMMER_GESTURE_CONFIG, useClass: IonicGestureConfig },
     OutcomeBehaviourMapProvider,
     LogHelper,
+    RekeySearchProvider,
```
Comment (TomBillingtonUK) [inline src/app/app.module.ts]: Does it need to be a global provider can it not be in the RekeySearch Page Module

**Review by TomBillingtonUK — APPROVED** [review]
Nice Work

File: src/app/app.module.ts (hunk 1 above)
Comment (szabikr) [reply src/app/app.module.ts]: Moved the declaration to the RekeySearchPage module.

File: src/providers/rekey-search/rekey-search.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+@Injectable()
+export class RekeySearchProvider {
+
+  constructor(
+    private httpClient: HttpClient,
+    private urlProvider: UrlProvider,
+  ) { }
+
+  getTest(rekeySearchParams: RekeySearchParams): Observable<any> {
```
Comment (roberthall12) [inline src/providers/rekey-search/rekey-search.ts]: Can this be a better type? `Observable<TestSlot>`?

File: src/pages/rekey-search/rekey-search.ts [hunk 3]
```diff
@@ -1,19 +1,33 @@
...
   templateUrl: 'rekey-search.html',
 })
-export class RekeySearchPage extends BasePageComponent {
+export class RekeySearchPage extends BasePageComponent implements OnInit {
+
+  pageState: RekeySearchPageState;
 
+  staffNumber: string = '';
+  applicationReference: string = '';
+  searchResults: any[] = [];
```
Comment (roberthall12) [inline src/pages/rekey-search/rekey-search.ts]: Same again, could we use a more safe type?

**Review by roberthall12 — APPROVED** [review]
Looks good - would like to avoid using `any` type if possible

File: src/providers/rekey-search/rekey-search.ts (hunk 2 above)
Comment (szabikr) [reply src/providers/rekey-search/rekey-search.ts]: Added `TestSlot` type

File: src/pages/rekey-search/rekey-search.ts (hunk 3 above)
Comment (szabikr) [reply src/pages/rekey-search/rekey-search.ts]: Added `TestSlot` type

**Pull request merged**